docs/en_us/platform_api/source/courses/blocks.rst

```diff
@@ -0,0 +1,347 @@
+
+* depth: (integer or ``all``) Specify how far down in the course blocks
+  hierarchy to traverse down. A value of ``all`` specifies the entire
+  hierarchy. The default value is 0. Example: ``depth=all``.
+
+* requested_fields: (list) Indicates fields to return for each block, in
+  addition to id, type, and display_name, which are always returned. For the
+  list of possible fields, see the fields listed under ``blocks`` in the
+  **Response Values** section below. Example:
+  ``requested_fields=graded,format,student_view_multi_device``.
+
+* return_type: (string) Indicates in what data type to return the block data.
+  Supported values are ``dict`` and ``list``. The default value is ``dict``.
+
+* student_view_data: (list) Indicates for which block types the JSON data is
+  returned in the "student_view_data" response value. Example:
+  ``student_view_data=video``.
+
+* username: (string) Required, unless ``all_blocks`` is specified. Example:
+  ``username=anjali``.
+
+
+=====================
+Response Values
+=====================
+
+The following fields are returned with a successful response.
+
+* root: The ID of the root node of the requested course block structure.
+
+* blocks: A dictionary that maps block usage IDs to a collection of information
+  about each block. Each block contains the following fields.
+
+  * id: (string) The usage ID of the block.
+
+  * type: (string) The type of block. Possible values include course, chapter,
+    sequential, vertical, html, problem, video, and discussion. The type can
+    also be the name of a custom type of block.
+
+  * display_name: (string) The display name of the block.
+
+  * children: (list) If the block has child blocks, a list of IDs of the child
+    blocks. Returned only if "children" is included in the "requested_fields"
+    parameter.
+
+  * block_counts: (dict) For each block type specified in the block_counts
+    parameter to the endpoint, the aggregate number of blocks of that type for
+    this block and all of its descendants. Returned only if the "block_counts"
+    input parameter contains this block's type.
+
+  * graded (boolean) Whether or not the block or any of its descendants is
+    graded. Returned only if "graded" is included in the "requested_fields"
+    parameter.
+
+  * format: (string) The assignment type of the block. Possible values can be
+    "Homework", "Lab", "Midterm Exam", and "Final Exam". Returned only if
+    "format" is included in the "requested_fields" parameter.
+
+  * student_view_data: (dict) The JSON data for this block. Returned only if the
+    "student_view_data" input parameter contains this block's type.
+
+  * student_view_url: (string) The URL to retrieve the HTML rendering of this
+    block's student view. The HTML can include CSS and Javascript code. This
+    field can be used in combination with the ``student_view_multi_device``
+    field to decide whether to display this content to the user. This URL can
```

Comment:
> to decide whether to display this content to the user

It took me a while to understand this sentence. It seems mobile development centered.